api/types.go

```diff
@@ -592,13 +656,23 @@ func (pin Pin) Equals(pin2 Pin) bool {
 		return false
 	}
 
+	sort.Strings(pin1s.Parents)
```

Comment:
Move the parents tests to the last position, that way we don't need to do it if something else is different. Thus marginally cheaper.